openassessment/xblock/static/js/src/studio/oa_container.js

```diff
@@ -65,14 +65,19 @@ OpenAssessment.Container = function(ContainerItem, kwargs) {
     this.removeButtonClass = kwargs.removeButtonClass;
     this.containerItemClass = kwargs.containerItemClass;
     this.notifier = kwargs.notifier;
+    this.prepareElement = (typeof kwargs.prepareElement === 'function') ? kwargs.prepareElement : function(el) {
```

Comment:
@dmitry-viskov, Thanks for working with me on this. I suggest localizing tinyMCE dependencies to `Prompt`. Note that these suggestions reverse my previous comment about modifying `element` outside of `Prompt`. I'm sorry for flip-flopping – it is more important to contain dependencies.

1. In `OpenAssessment.Prompt.prototype`, create a new class property called `TinyMCEEnabled`. Compute `tinyMCEEnabled` as `window.tinyMCE !== undefined`.
2. Move the `prepareElement` definition to `OpenAssessment.Prompt.prototype`. Rename the function as `attachWysiwygToPrompt`. At the top of the function definition, assert that `self.tinyMCEEnabled` is true (the assertion enforces that tinyMCE must be available for the function to be usable).
3. In the `OpenAssessment.Prompt` constructor, assign `self.element` in the following way: 
```
if (self.tinyMCEEnabled)
{
    element = self.attachWysiwygToPrompt(element);
}
self.element = element;
```